module/zfs/spa.c

```diff
@@ -4650,6 +4805,145 @@ spa_ld_check_features(spa_t *spa, boolean_t *missing_feat_writep)
 	return (0);
 }
 
+struct load_chain_map_arg {
+	blkptr_t *bp;
+	spa_t *spa;
+	uint64_t *error;
+};
+
+static int
+load_chain_map_claim_blk_cb(spa_t *spa, const blkptr_t *bp, void *arg)
+{
+	(void) arg;
+	int error = metaslab_claim(spa, bp,
+	    spa_get_dsl(spa)->dp_tx.tx_open_txg);
+	if (error == ENOENT)
+		error = 0;
+	return (error);
+}
+
+static int
+load_chain_map_claim_lr_cb(spa_t *spa, const lr_t *lrc, void *arg)
+{
+	(void) arg;
+	lr_write_t *lr = (lr_write_t *)lrc;
+	blkptr_t *bp = &lr->lr_blkptr;
+	if (lrc->lrc_txtype != TX_WRITE || BP_IS_HOLE(bp))
```

Comment:
It seems you are duplicating normal ZIL claiming logic here, and that normal logic is actually now handles also `TX_CLONE_RANGE`, that you seems not.

Reply:
This was written before TX_CLONE_RANGE, so that makes sense. The fix for this is to do a claim pass on client pools too, but only handle TX_CLONE_RANGE, since the CLONE_RANGE logic can only be done with the client pool's context.